code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -384,4 +409,22 @@ private Map<String, Object> prepareConsentXDMMapWithPayload(final Map<String, Ob
 		consentMap.put(ConsentConstants.EventDataKey.CONSENTS, payload);
 		return consentMap;
 	}
+
+	/**
+	 * Determines if the SDK should force a sync of the consent preferences to the Edge Network
+	 * based on the forceSync flag and the timestamp of the event that triggered the consents update,
+	 * regardless if the consent preferences changed.
+	 *
+	 * @param event the {@link Event} that triggered the consents update.
+	 * @return true if the SDK should force a sync, false otherwise.
+	 */
+	private boolean shouldForceSync(final Event event) {
```

Comment:
Same concern as iOS why do we check for time. when I am thinking of forceSync, it should go out no matter what. Adding more conditions like this would just make one more branch in the logic.

Reply:
The forceSync flag will essentially revert the behavior to the current, which is to send a Profile update every time a Consent update request is made, regardless if the values change. This option is needed for some customers who may need to send Profile updates to different sandboxes. The time check is an added optimization to ignore quick, repeated, duplicate requests which are viewed as an error in the application implementation.